src/libostree/ostree-repo-commit.c

```diff
@@ -848,65 +848,57 @@ devino_equal (gconstpointer   a,
 }
 
 static gboolean
-scan_loose_devino (OstreeRepo                     *self,
-                   GHashTable                     *devino_cache,
-                   GCancellable                   *cancellable,
-                   GError                        **error)
+scan_one_loose_devino (OstreeRepo                     *self,
+                       int                             object_dir_fd,
+                       GHashTable                     *devino_cache,
+                       GCancellable                   *cancellable,
+                       GError                        **error)
 {
   gboolean ret = FALSE;
-  guint i;
-  OstreeRepoMode repo_mode;
-  gs_unref_ptrarray GPtrArray *object_dirs = NULL;
-
-  if (self->parent_repo)
-    {
-      if (!scan_loose_devino (self->parent_repo, devino_cache, cancellable, error))
-        goto out;
-    }
-
-  repo_mode = ostree_repo_get_mode (self);
+  int res;
+  g_auto(GLnxDirFdIterator) dfd_iter = { 0, };
 
-  if (!_ostree_repo_get_loose_object_dirs (self, &object_dirs, cancellable, error))
+  if (!glnx_dirfd_iterator_init_at (object_dir_fd, ".", FALSE,
+                                    &dfd_iter, error))
     goto out;
-
-  for (i = 0; i < object_dirs->len; i++)
+  
+  while (TRUE)
     {
-      GFile *objdir = object_dirs->pdata[i];
-      gs_unref_object GFileEnumerator *enumerator = NULL;
-      gs_unref_object GFileInfo *file_info = NULL;
-      const char *dirname;
+      struct dirent *dent;
+      g_auto(GLnxDirFdIterator) child_dfd_iter = { 0, };
 
-      enumerator = g_file_enumerate_children (objdir, OSTREE_GIO_FAST_QUERYINFO,
-                                              G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
-                                              cancellable,
-                                              error);
-      if (!enumerator)
+      if (!glnx_dirfd_iterator_next_dent (&dfd_iter, &dent, cancellable, error))
         goto out;
+          
+      if (dent == NULL)
+        break;
 
-      dirname = gs_file_get_basename_cached (objdir);
+      if (strlen (dent->d_name) != 2)
```

Comment:
This `strlen` check could use a comment.  I think it's to recurse only in the two-digit hex directories, right?